mmengine/hooks/checkpoint_hook.py

```diff
@@ -74,6 +74,12 @@ class CheckpointHook(Hook):
         file_client_args (dict, optional): Arguments to instantiate a
             FileClient. See :class:`mmcv.fileio.FileClient` for details.
             Defaults to None.
+        filename_tmpl (str, optional): String template to indicate checkpoint
```

Comment:
Is there any chance that this filename needs to be set manually?
If not, we may consider not making it an argument for now, as this might affect other places related to checkpoint loading.

Reply:
Seems like some developers are currently using this feature, so I think it's good to make it public. This change does not affect those repos not using this feature, since default behavior stays unchanged.